src/main/java/seedu/address/model/task/UniqueTaskList.java

```diff
@@ -49,6 +49,13 @@ public void add(Task toAdd) {
         internalList.add(toAdd);
     }
 
+    /**
+     * Sorts the tasks according to deadline.
+     */
+    public void sortTasks() {
+        internalList.sort((a, b) -> a.compareTo(b));
+    }
+
```

Comment:
Actually since you already specify the compareTo method, the comparator inside sort is redundant.